opacus/grad_sample/conv.py

```diff
@@ -1,16 +1,52 @@
 #!/usr/bin/env python3
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 
-from typing import Dict, Union
+from typing import Dict, Union, Tuple
 
 import numpy as np
+import time
 import torch
 import torch.nn as nn
+import torch.nn.functional as F
 from opacus.utils.tensor_utils import unfold3d
 
 from .utils import register_grad_sampler
 
 
+def unfold2d(
+    input,
+    kernel_size: Tuple[int, int],
+    padding: Tuple[int, int],
+    stride: Tuple[int, int],
+    dilation: Tuple[int, int],
+):
+    # print("input shape", input.shape)
+    *shape, H, W = input.shape
+    H_effective = (
+        H + 2 * padding[0] - (kernel_size[0] + (kernel_size[0] - 1) * (dilation[0] - 1))
+    ) // stride[0] + 1
```

Comment:
Me trying to visualise the shape:
![giphy](https://user-images.githubusercontent.com/1632501/141211657-f7c1643d-dae3-4bd3-890c-de1323f52904.gif)

Reply:
Should I comment/document?

Reply:
I find tensor shapes as the most useful comments to help understand this stuff, anything beyond is probably too much.

Reply:
+1 commenting with the shapes should be sufficient. If you can easily explain in a line, that works too; but a paragraph is not worth it IMO.